tests/Feature/BookTest.php

```diff
@@ -44,6 +45,53 @@ public function test_get_all_books(): void
                 });
         });
 }
+    public function test_get_all_claimed_books(): void
+    {
+        Claim::factory()->create();
+
+        $response = $this->getJson('/api/books?claimed=1');
+
+        $response
+                ->assertStatus(200)
+                ->assertJson(function(AssertableJson $json) {
+                    $json
+                        ->hasAll(['message', 'data'])
+                        ->has('data', 1, function(AssertableJson $json) {
+                            $json
+                                ->hasAll (['id', 'title', 'author','blurb', 'page_count', 'year', 'image', 'created_at', 'updated_at', 'genre_id', 'genre', 'reviews', 'claimed_by_name',]);
+                                    
+                });
+        });
+
+    }
+
+    public function test_get_all_unclaimed_books(): void
+    {
+        $book = Book::factory()->create();
+        $claim = Claim::factory()->create();
+       
+        $book->claimed_by_name = '';
+        $claim->book_id = '';
+      
+        $book->save();
+        $claim->save();
+
+        $response = $this->getJson('/api/books?claimed=0');
+
+        $response
+                ->assertStatus(200)
+                ->assertJson(function(AssertableJson $json) {
+                    $json
+                        ->hasAll(['message', 'data'])
+                        ->has('data', 1, function(AssertableJson $json) {
+                            $json
+                                ->hasAll (['id', 'title', 'author','blurb', 'page_count', 'year', 'image', 'created_at', 'updated_at', 'genre_id', 'genre', 'reviews', 'claimed_by_name',]);
+                                    
+                });
+        });
+
+    }
```

Comment:
Once you have added the validation I mentioned, you should then add a test that proves it works by sending a request like ?claimed=9 or something similar

Reply:
Thanks, test_get_all_claimed_books_invalid() added to test suite.